src/extensions/snippets/SnippetsView/SnippetsWidget.tsx

```diff
@@ -13,18 +14,34 @@ type Props = {
     height: number
 }
 
-const SnippetsWidget: FunctionComponent<Props> = ({ recording, sorting, selection, selectionDispatch, unitIds, width }) => {
+const WhenVisible: FunctionComponent<{width: number, height: number}> = ({width, height, children}) => {
```

Comment:
This WhenVisible is a convenient wrapper component that hides its child until it has come visible (via scrolling, eg) for the first time.